src/Common/src/Interop/Windows/winhttp/Interop.SafeWinHttpHandle.cs

```diff
@@ -45,12 +64,8 @@ internal sealed class SafeWinHttpHandleWithCallback : SafeWinHttpHandle
             // Add a reference to this object so that the AppDomain doesn't get unloaded before the last WinHttp callback.
             public void AttachCallback()
             {
-                bool success = false;
-                DangerousAddRef(ref success);
-                if (!success)
-                {
-                    throw new InvalidOperationException("Could not increase SafeHandle reference.");
-                }
+                bool ignore = false;
```

Comment:
Yes, the documentation is incorrect:

See previous response from @jkotas:

>
This weird pattern was invented for thread abort hardening (on desktop). Thread abort exception can be injected anywhere. The "ref bool success" was intended to make it easy for you cleanup code to determine whether to call DangerousRelease.
The documentation for Monitor.Enter(object, ref bool) - that uses the same pattern - has more detailed description.
•	The check for !success in this code can be deleted - it won't be ever executed.
•	You do not need to do anything else since we are not doing thread abort hardening in .NET Core. If you would like to write it hardened, you would need to add bool field to your object; and cleanup in finalizer only if it is set.